plugin/storage/cassandra/spanstore/operation_names_test.go

```diff
@@ -108,38 +138,49 @@ func TestOperationNamesStorageWrite(t *testing.T) {
 
 func TestOperationNamesStorageGetServices(t *testing.T) {
 	var scanError = errors.New("scan error")
-	var writeCacheTTL time.Duration
-	var matched bool
-	matchOnce := mock.MatchedBy(func(v []interface{}) bool {
-		if matched {
-			return false
-		}
-		matched = true
-		return true
-	})
-	matchEverything := mock.MatchedBy(func(v []interface{}) bool { return true })
-	for _, expErr := range []error{nil, scanError} {
-		withOperationNamesStorage(writeCacheTTL, func(s *operationNameStorageTest) {
-			iter := &mocks.Iterator{}
-			iter.On("Scan", matchOnce).Return(true)
-			iter.On("Scan", matchEverything).Return(false) // false to stop the loop
-			iter.On("Close").Return(expErr)
-
-			query := &mocks.Query{}
-			query.On("Iter").Return(iter)
-
-			s.session.On("Query", mock.AnythingOfType("string"), []interface{}{"service-a"}).Return(query)
-
-			services, err := s.storage.GetOperations("service-a")
-			if expErr == nil {
-				assert.NoError(t, err)
-				// expect empty string because mock iter.Scan(&placeholder) does not write to `placeholder`
-				assert.Equal(t, []string{""}, services)
-			} else {
-				assert.EqualError(t, err, "Error reading operation_names from storage: "+expErr.Error())
-			}
-		})
+	for _, test := range []struct {
+		name          string
+		schemaVersion schemaVersion
+		expErr        error
+	}{
+		{name: "test old schema without error", schemaVersion: previousVersion, expErr: nil},
+		{name: "test old schema with scan error", schemaVersion: previousVersion, expErr: scanError},
+		{name: "test new schema without error", schemaVersion: latestVersion, expErr: nil},
+		{name: "test new schema with scan error", schemaVersion: latestVersion, expErr: scanError},
+	} {
+		t.Run(fmt.Sprintf("%s", test.name), func(t *testing.T) {
```

Comment:
I have optimized to use t.Run(test.name, func) in the PR 1921-3. Try to avoid update it in 2nd branch and merge into 3rd branch again.